subprojects/gradle/module-dependencies-graph/src/main/kotlin/com/avito/module/internal/dependencies/ModuleProjectConfigurationDependenciesNode.kt

```diff
@@ -17,15 +17,16 @@ internal class ModuleProjectConfigurationDependenciesNode(
     }
 
     fun allDependencies(): Set<Project> {
-        return dependencies.mapValues { (_, nodeSet) ->
-            nodeSet.flatMap { node ->
-                mutableSetOf(node.project).also {
-                    it.addAll(
-                        node.allDependencies()
```

Comment:
Recursive `allDependencies` has terrible complexity. 
It repeatedly collects all dependencies in each node.